### PR TITLE
switched PodDisruptionBudget from v1beta1 to v1 in time for 1.25

### LINKED ISSUE #4285: policy/v1beta1 PodDisruptionBudget is deprecated in v1.21+, unavailable in v1.25+
**Which component are you using?**: cluster-autoscaler

**What version of the component are you using?**: 

Component version: v1.21.0, chart 9.10.5

**What k8s version are you using (`kubectl version`)?**:

<details><summary><code>kubectl version</code> Output</summary><br><pre>
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.1", GitCommit:"632ed300f2c34f6d6d15ca4cef3d3c7073412212", GitTreeState:"clean", BuildDate:"2021-08-19T15:38:26Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"21+", GitVersion:"v1.21.2-eks-0389ca3", GitCommit:"8a4e27b9d88142bbdd21b997b532eb6d493df6d2", GitTreeState:"clean", BuildDate:"2021-07-31T01:34:46Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}
</pre></details>

**What environment is this in?**:

I use Flux/Helm Operator to deploy the chart.

**What did you expect to happen?**:

I expect the chart to use, when possible, the latest versions of the APIs or to use the appropriate APIs based on the Kubernetes version of the cluster. 

**What happened instead?**:

```shell
W0826 12:53:50.687259       6 warnings.go:70] policy/v1beta1 PodDisruptionBudget is deprecated in v1.21+, unavailable in v1.25+; use policy/v1 PodDisruptionBudget
W0826 12:53:50.692111       6 warnings.go:70] policy/v1beta1 PodDisruptionBudget is deprecated in v1.21+, unavailable in v1.25+; use policy/v1 PodDisruptionBudget
W0826 12:53:50.712486       6 warnings.go:70] policy/v1beta1 PodDisruptionBudget is deprecated in v1.21+, unavailable in v1.25+; use policy/v1 PodDisruptionBudget
```

**How to reproduce it (as minimally and precisely as possible)**:

Install the helm chart in a cluster with K8s > 1.20 

**Anything else we need to know?**: 

Nothing

### PR DESCRIPTION
**Pull request by mikelo**
#### Which component this PR applies to?

cluster-autoscaler

#### What type of PR is this?

/kind api-change

#### Which issue(s) this PR fixes:

Fixes #4285

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster-autoscaler/utils/drain/drain.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ import (
 	"time"
 
 	apiv1 "k8s.io/api/core/v1"
-	policyv1 "k8s.io/api/policy/v1beta1"
+	policyv1 "k8s.io/api/policy/v1"
```
Comment (yaroslava-serdiuk) [inline cluster-autoscaler/utils/drain/drain.go:24]: I suppose the same with  checkKubeSystemPDBs() here

File: cluster-autoscaler/simulator/drain.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ import (
 	"time"
 
 	apiv1 "k8s.io/api/core/v1"
-	policyv1 "k8s.io/api/policy/v1beta1"
+	policyv1 "k8s.io/api/policy/v1"
```
Comment (yaroslava-serdiuk) [inline cluster-autoscaler/simulator/drain.go:24]: Having the changes between v1 and v1beta1 https://kubernetes.io/docs/reference/using-api/deprecation-guide/#poddisruptionbudget-v125 seems we need to add check for {} selector in  checkPdbs.

**Comment by liggitt** [comment]
cc @yaroslava-serdiuk

**Review by yaroslava-serdiuk — CHANGES_REQUESTED** [review]
I don't think we need 3 commits for this change, can you make everything in one commit, please?

File: cluster-autoscaler/simulator/drain.go (hunk 2 above)
Comment (yaroslava-serdiuk) [reply cluster-autoscaler/simulator/drain.go:24]: Also we need add a test case for empty selector.

Comment (liggitt) [reply cluster-autoscaler/simulator/drain.go:24]: actually, checkPdbs is already using metav1.LabelSelectorAsSelector(pdb.Spec.Selector), which means a null selector selects nothing and an empty selector `{}` selects everything

that means CA was treating v1beta1 PDBs incorrectly, and is already following v1 semantics, so no changes are needed

Comment (liggitt) [reply cluster-autoscaler/simulator/drain.go:24]: > Also we need add a test case for empty selector.

I agree with that (and for null selector)

**Comment by mikelo** [comment]
> I don't think we need 3 commits for this change, can you make everything in one commit, please?

yeah will

> I don't think we need 3 commits for this change, can you make everything in one commit, please?

yes will fix that ASAP
